### PR TITLE
Update List of features for Godot 4.0.rc1

### PR DESCRIPTION
**Pull request by Calinou**
This also adds more links to documentation within the page.

### CODE REVIEW

File: about/list_of_features.rst
Comment (Piralein) [inline about/list_of_features.rst:120]:
```suggestion
- :ref:`Font rendering <doc_gui_using_fonts>` using bitmaps, rasterization using FreeType
```

File: about/list_of_features.rst
Comment (Piralein) [inline about/list_of_features.rst:541]:
```suggestion
  :ref:`2D <doc_navigation_overview_2d>` and :ref:`3D <doc_navigation_overview_3d>`.
```

File: about/list_of_features.rst
Comment (Piralein) [inline about/list_of_features.rst:178]: The reference `doc_variable_rate_shading` is invalid, not sure what to link here.

File: about/list_of_features.rst
Comment (Piralein) [inline about/list_of_features.rst:609]:
```suggestion
  and :ref:`iOS <doc_plugins_for_ios>`.
```

File: about/list_of_features.rst
Comment (Calinou) [reply about/list_of_features.rst:178]: This is pending on https://github.com/godotengine/godot-docs/pull/6402 to be merged :slightly_smiling_face:

**Review by YuriSizov — APPROVED** [review]
LGTM, thanks for proofreading it! Though we cannot merge until the referenced PR is merged, or until the text here is temporarily adjusted to remove the mention/link.

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**
